### PR TITLE
add OnNewLeader callback to leaderelection client

### PR DESCRIPTION
**Pull request by mikedanese**
@davidopp @timothysc

### CODE REVIEW

File: pkg/client/leaderelection/leaderelection_test.go [hunk 1]
```diff
@@ -237,5 +248,14 @@ func TestTryAcquireOrRenew(t *testing.T) {
 		if !test.transitionLeader && le.observedRecord.LeaderTransitions != 0 {
 			t.Errorf("[%v]leader should not have transitioned but did", i)
 		}
+		if !test.transitionLeader && le.observedRecord.LeaderTransitions != 0 {
```
Comment (davidopp) [inline pkg/client/leaderelection/leaderelection_test.go]: This block appears to be identical to the previous block, unless I'm mis-reading.

Comment (mikedanese) [inline pkg/client/leaderelection/leaderelection_test.go]: Oops. Fixed.

**Comment by mikedanese** [comment]
@k8s-bot unit test this please

**Comment by davidopp** [comment]
LGTM

**Comment by mikedanese** [comment]
@k8s-bot unit test this please

**Pull request merged**